### PR TITLE
0.9 compatibility - version HTTP headers, fallbackHosts, request() and decoding interop. test

### PR DESCRIPTION
**Pull request by bladeSk**

### CODE REVIEW

File: src/Http.php [hunk 1]
```diff
@@ -117,7 +119,7 @@ public function request( $method, $url, $headers = array(), $params = array() )
                 $this->curl->setOpt( $ch, CURLOPT_POSTFIELDS, $params );
 
                 if ($this->postDataFormat == 'json') {
-                    array_push( $headers, 'Accept: application/json', 'Content-Type: application/json' );
+                    array_push( $headers, 'Content-Type: application/json' );
```
Comment (mattheworiordan) [inline src/Http.php]: Out of interest, why did you remove the Accept header.  We default to JSON in our response, but isn't it better to be explicit?

File: tests/AblyRestTest.php [hunk 2]
```diff
@@ -222,7 +222,7 @@ public function testCustomHostAndFallbacks() {
             $this->fail('Expected the request to fail');
         } catch(AblyRequestException $e) {
            $this->assertEquals( $hostWithFallbacks[0], $ably->http->failedHosts[0], 'Expected to try restHost first' );
-            $this->assertNotEquals( $hostWithFallbacks, $ably->http->failedHosts, 'Expected to have fallback hosts randomized' );
+            // $this->assertNotEquals( $hostWithFallbacks, $ably->http->failedHosts, 'Expected to have fallback hosts randomized' ); // this may fail when randomized order matches the original order
```
Comment (mattheworiordan) [inline tests/AblyRestTest.php]: Yup, we have approached this problem quite differently in other langauges, often just checking that the random method is just called.

File: src/Models/Response.php [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            }
+
+            if ( isset($this->headers['X-Ably-Errormessage']) ) {
+                $this->errorMessage = $this->headers['X-Ably-Errormessage'];
+            }
+        } else {
+            $this->success = true;
+        }
+
+        $this->ably = null; // !!!!!!!!!!!!!!!!!!!!! remove
```
Comment (mattheworiordan) [inline src/Models/Response.php]: What is this comment for?

File: tests/ChannelMessagesTest.php [hunk 4]
```diff
@@ -409,6 +409,31 @@ public function testMessageEncodings() {
...
+            $ably->channel("testInterop_$i")->publish($msg);
+            $ret = $ably->channel("testInterop_$i")->history();
+
+            if ( isset( $testMsgData->expectedHexValue ) ) {
+                $expected = pack( 'H*', $testMsgData->expectedHexValue );
+            } else {
+                $expected = $testMsgData->expectedValue;
+            }
+
+            $this->assertEquals($expected, $ret->items[0]->data);
```
Comment (mattheworiordan) [inline tests/ChannelMessagesTest.php]: Unfortunately this is not doing what is required, see https://github.com/ably/ably-ruby/pull/94 and https://github.com/ably/ably-ios/pull/459 for example.  You need to publish a raw message using a raw HTTP lib (i.e. not through the Ably client lib) and then check that using history it's decoded as expected, and then publish using the Ably client lib, and use history using a RAW http lib to check that the JSON matches what is expected.

**Comment by mattheworiordan** [comment]
Looks good, but unfortunately the interoperability test is not doing enough. See https://github.com/ably/docs/pull/146 as well

File: src/Http.php (hunk 1 above)
Comment (bladeSk) [inline src/Http.php]: The accept header was sent only when there were either GET parameters or POST body set. This was a wrong place to set the header. It should be probably sent in all the requests, I'll fix that.

File: src/Models/Response.php (hunk 3 above)
Comment (bladeSk) [inline src/Models/Response.php]: Oh crap, I was supposed to remove that, thanks for noticing ;)

**Comment by mattheworiordan** [comment]
Nice work @bladeSk, that looks great :)
